### ISSUE TITLE
Hyperparameter Tuning

### ISSUE THREAD

**Issue by aaprasad** [lindermanlab/ppseq-pytorch#1]
Hi there! Thanks for the tool, it's super straightforward to use. I just had a question on how to select the hyperparameters for the model. Specifically, a description of each hyperparameter and any advice on how to choose the hyperparameters. I assume that `num_templates` refers to the max number of sequences to search for and `template_duration` is the length of the sequence in `num_timepoints`. Also, I noticed you all mentioned that a notebook for cross-validation would be out soon? Is there any update on that. It would be nice in case I need to just run a hyperparameter sweep on the model for my data. Thanks for the help in advanced!

**Comment by li-hong-yue** [comment]
Hi Mr. Aaditya Prasad,

For the parameters, 
'num_templates' refers to the number of sequences to search for; 
'template_duration' refers to the 'delay window'. Basically, given a stimulus, how long it will last. It is the width of the W1/W2 in the first figure in the songbird notebook.

For the hyperparameters, 'alpha_*', 'beta_*' are the priors of gamma distribution for the amplitudes, templates, and background noise rate. I recommend to set alpha = 1 and beta = 0 to start with.
 
For validation, 
https://colab.research.google.com/drive/1Ej8Gpqg4EUdwMrBMbBAgyV36ilAXuoXQ?usp=sharing 
This is a notebook that I’ve used to mask out the data, train a model, and then evaluate log likelihood on the masked data.
Basically, the higher the log-likelihood on the masked data, the better the fitting.
Hope it is helpful!

Best,
PPSeq Developers
